src/ORM/UniqueKey/UniqueKeyInterface.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SilverStripe\ORM\UniqueKey;
+
+use SilverStripe\ORM\DataObject;
+
+interface UniqueKeyInterface
+{
+    /**
+     * Generate a unique key for data object
+     *
+     * @param DataObject $object
+     * @param array $extraKeys
+     * @return string
+     */
+    public function generateKey(DataObject $object, array $extraKeys = []): string;
```

Comment:
$extraKeys isn't great naming, you've referred to it as $cacheKeys just above in DataObject, no reason why these variable names shouldn't match